src/mame/skeleton/cubieboard4.cpp

```diff
@@ -0,0 +1,127 @@
+// license:BSD-3-Clause
+// copyright-holders:
+/***********************************************************************************************
+
+Skeleton driver for "Cubieboard4" (CC A-80) and arcade games based on it.
+
+Cubieboard4, also named CC-A80, is a open source mini PC or single board computer.
+The main chipset Allwinner A80 is a 28nm Octa-Core A15/A7 big.LITTLE architecture application processor
+with a CPU dominant frequency of 2GHz. It also has the GPU 64-core GPU graphics core PowerVR G6230 which
+supports openGL ES, OpenGL, and OpenCL.
+CC-A80 has the standard interfaces like desktop computer, such as HDMI & VGA, 100M/1000M RJ45, 4 USB2.0
+host ports, 1 USB3.0 OTG port, audio output, microphone input, dual-band wifi and bluetooth 4.0, micro SD
+card. It has 2GB DDR3 on-board memory and support Li-Po battery UPS powerinput.
+
+Main Cubieboard4 components:
+-Allwinner A80.
+-Realtek RTL8211E integrated Ethernet transceiver.
+-AMPAK Technology AP6330 WiFi + Bluetooth 4.0(HS) + FM Rx Module.
+-SKhynix KLM4G1YEMD-B031 (embedded MultiMediaCard Ver. 5.0 compatible, soldered to the back side of the PCB).
+-4 x H5TQ4G63AFR 4Gb DDR3 SDRAM.
+-X-Powers AXP809 PMIC.
+-X-Powers AC100 audio codec and RTC subsystem.
+
+***********************************************************************************************/
+
+#include "emu.h"
+#include "cpu/arm7/arm7.h"
+#include "bus/generic/slot.h"
+#include "bus/generic/carts.h"
+#include "softlist_dev.h"
+
+namespace {
+
+class cubiecca80_state : public driver_device
+{
+public:
+	cubiecca80_state(const machine_config &mconfig, device_type type, const char *tag) :
+		driver_device(mconfig, type, tag),
+		m_cart(*this, "cartslot"),
+		m_maincpu(*this, "maincpu")
+
+	{ }
+
+	void cubiecca80(machine_config &config);
+
+protected:
+	optional_device<generic_slot_device> m_cart;
+
+private:
+	required_device<cpu_device> m_maincpu;
+};
+
+static INPUT_PORTS_START( cubiecca80 )
+INPUT_PORTS_END
+
+void cubiecca80_state::cubiecca80(machine_config &config)
+{
+	// Basic machine hardware
+	ARM9(config, m_maincpu, 200'000'000); // Actually an Allwinner A80 2 GHz
+
+	// Video hardware
+	//SCREEN(...)
+
+	// Audio hardware
+	//SPEAKER(...)
+
+	GENERIC_CARTSLOT(config, m_cart, generic_plain_slot, "cubieboard4_sdcard");
```

Comment:
As above, needs to be shorthanded to `	GENERIC_CARTSLOT(config, m_cart_ext, generic_plain_slot, "sdcard");` which also hints being the wrong device type here (question arises about how `SPI_SDCARD` device is supposed to load from a SW interface so I don't mind for that atm).
Another one needs to be added above here, `GENERIC_CARTSLOT(config, m_cart_int, generic_plain_slot, "emmc");`, so that you can move files below as `interface="emmc"`.

Reply:
Done!!
Non-arcade software moving to the software list, having two interfaces, "sdcard" and "emmc".

Thanks!!!!!